src/PositionManager.sol

```diff
@@ -171,36 +153,30 @@ contract PositionManager is
         } else if (currencyDelta > 0) {
             _take(currency, caller, uint256(currencyDelta));
         }
-
-        return abi.encode(currencyDelta);
     }
 
     /// @param params is an encoding of uint256 tokenId, PositionConfig memory config, bytes hookData
     /// @dev this is overloaded with ERC721Permit._burn
-    function _burn(bytes memory params) internal returns (bytes memory) {
+    function _burn(bytes memory params) internal {
         (uint256 tokenId, PositionConfig memory config, bytes memory hookData) =
             abi.decode(params, (uint256, PositionConfig, bytes));
 
-        if (!_isApprovedOrOwner(_getLocker(), tokenId)) revert NotApproved(_getLocker());
+        if (!_isApprovedOrOwner(_msgSender(), tokenId)) revert NotApproved(_msgSender());
         if (positionConfigs[tokenId] != config.toId()) revert IncorrectPositionConfigForTokenId(tokenId);
         uint256 liquidity = uint256(_getPositionLiquidity(config, tokenId));
 
         // Can only call modify if there is non zero liquidity.
-        BalanceDelta delta;
-
-        if (liquidity > 0) delta = _modifyLiquidity(config, -(liquidity.toInt256()), bytes32(tokenId), hookData);
+        if (liquidity > 0) _modifyLiquidity(config, -(liquidity.toInt256()), bytes32(tokenId), hookData);
 
         delete positionConfigs[tokenId];
         // Burn the token.
         _burn(tokenId);
-        return abi.encode(delta);
     }
 
     function _modifyLiquidity(PositionConfig memory config, int256 liquidityChange, bytes32 salt, bytes memory hookData)
         internal
-        returns (BalanceDelta liquidityDelta)
```

Comment:
we still need to return liquidityDelta because of slippage checks inside the internal functions

unless slippage should be on the CLOSE command 🤔